test/success/connection.prl

```diff
@@ -7,13 +7,20 @@ Thm Connection(#A) : [
 ] by [
   {
   lam a. lam b. lam p.
-    <i> <j> 'hcom{i=0,i=1,j=0,j=1;0~>1}(#A; ,a;
-     {k}.hcom{k=0,k=1;1~>j}(#A; hcom{k=0,k=1;1~>0}(#A; ,p @ k; {_}.,a; {j}.,p @ j); {_}.,a; {_}.,a);
-     {k}.hcom{k=0,k=1;1~>j}(#A; ,p @ k; {_}.,a; {j}.,p @ j);
-     {k}.hcom{k=0,k=1;1~>i}(#A; hcom{k=0,k=1;1~>0}(#A; ,p @ k; {_}.,a; {i}.,p @ i); {_}.,a; {_}.,a);
-     {k}.hcom{k=0,k=1;1~>i}(#A; ,p @ k; {_}.,a; {j}.,p @ j))
+    <i> <j> 
+      'hcom{0~>1}(#A; ,a)
+        [i=0 -> {k}.
+          hcom{1~>j}(#A; hcom{1~>0}(#A; ,p @ k) [k=0 -> {_}.,a; k=1 -> {l}.,p @ l])
+            [k=0 -> {_}.,a;
+             k=1 -> {_}.,a];
+         i=1 -> {k}. hcom{1~>j}(#A; ,p @ k) [k=0 -> {_}.,a; k=1 -> {l}.,p @ l];
+         j=0 -> {k}.
+          hcom{1~>i}(#A; hcom{1~>0}(#A; ,p @ k) [k=0 -> {_}.,a; k=1 -> {l}.,p @ l])
+            [k=0 -> {_}.,a;
+             k=1 -> {_}.,a];
+         j=1 -> {k}. hcom{1~>i}(#A; ,p @ k) [k=0 -> {_}.,a; k=1 -> {l}.,p @ l]]
   };
-  auto;
+  auto
```

Comment:
BTW, I tested this file with both SML/NJ and MLton; in SML/NJ it takes .76s to refine, whereas with MLton it takes only .46s to refine.